don't suggest borrowing the `Borrow::borrow` function item

### Code

```Rust
use std::borrow::Borrow;
const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
```


### Current output

```Shell
Compiling playground v0.0.1 (/playground)
error[E0277]: the trait bound `&'static String: Borrow<str>` is not satisfied
 --> src/lib.rs:2:55
  |
2 | const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
  |                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Borrow<str>` is not implemented for `&'static String`
  |
help: consider borrowing here
  |
2 | const F: for<'a> fn(&'a &'static String) -> &'a str = &<&'static String as Borrow<str>>::borrow;
  |                                                       +

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground` due to previous error
```


### Desired output

```Shell
Compiling playground v0.0.1 (/playground)
error[E0277]: the trait bound `&'static String: Borrow<str>` is not satisfied
 --> src/lib.rs:2:55
  |
2 | const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
  |                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Borrow<str>` is not implemented for `&'static String`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground` due to previous error
```


### Rationale and extra context

borrowing the `Borrow::borrow` function item won't ever fix the missing trait bound.

### Other cases

_No response_

### Anything else?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=936662eafc8d3c0e04f880536d242015

--- a/tests/ui/traits/suggest-remove-reference-in-where-clause.rs
+++ b/tests/ui/traits/suggest-remove-reference-in-where-clause.rs
@@ -1,0 +1,4 @@
+use std::borrow::Borrow;
+pub const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
+//~^ ERROR E0277
+fn main() {}

--- a/tests/ui/traits/suggest-remove-reference-in-where-clause.stderr
+++ b/tests/ui/traits/suggest-remove-reference-in-where-clause.stderr
@@ -1,0 +1,15 @@
+error[E0277]: the trait bound `&'static String: Borrow<str>` is not satisfied
+  --> $DIR/suggest-remove-reference-in-where-clause.rs:2:60
+   |
+LL | pub const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
+   |                                                            ^^^^^^^^^^^^^^^ the trait `Borrow<str>` is not implemented for `&'static String`
+   |
+help: consider removing the leading `&`-reference
+   |
+LL - pub const F: for<'a> fn(&'a &'static String) -> &'a str = <&'static String as Borrow<str>>::borrow;
+LL + pub const F: for<'a> fn(&'a &'static String) -> &'a str = <String as Borrow<str>>::borrow;
+   |
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0277`.

